6.5.0: release notes (#3394)
sensu/sensu-docs: content/sensu-go/6.5/release-notes.md

Read the [upgrade guide][1] for information about upgrading to the latest version of Sensu Go.

---
## 6.5.0 release notes

**October 12, 2021** &mdash; The latest release of Sensu Go, version 6.5.0, is now available for download.

Insert Release Synopsis.

Read the [upgrade guide][1] to upgrade Sensu to version 6.5.0.

**NEW FEATURES:**

> - Added core/v2.Pipeline resource for configuring event pipelines.
> - Added sensuctl commands for pipeline list, info, and delete.

Reviewer: I think you can combine these two items.

```suggestion
- New [pipelines][] resource allows you to specify event filters, mutators, and handlers in a single workflow instead of listing filters and mutators in handler definitions. You can reference pipelines in your check definitions. The [/pipelines API endpoint][] provides HTTP access for retrieving pipeline data and configuring pipelines, and you can use [sensuctl][] to manage pipelines.
```

Reviewer: The three links to add for this item are:
https://docs.sensu.io/sensu-go/6.5/observability-pipeline/observe-process/pipelines/
https://docs.sensu.io/sensu-go/6.5/api/pipeline-resource/
https://docs.sensu.io/sensu-go/6.5/sensuctl/create-manage-resources/#manage-resources

> - Added pipelines field to Check and CheckConfig

Reviewer: ```suggestion
- Check definitions now include the [pipelines attribute][] for specifying pipeline resources to use for the check's observability events.
```

The link for this one is https://docs.sensu.io/sensu-go/6.5/observability-pipeline/observe-schedule/checks/#pipelines-attribute

> **FIXES:**

Reviewer: Fixes should be listed last (new features, improvements, fixes)

> - Sensuctl env now properly displays the SENSU_API_KEY and SENSU_TIMEOUT environment variables.

Reviewer: ```suggestion
- sensuctl env now properly lists `SENSU_API_KEY` and `SENSU_TIMEOUT` as available options for [environment variable export][].
```

The link for this one is https://docs.sensu.io/sensu-go/6.5/sensuctl/environment-variables/#export-environment-variables-with-sensuctl-env

> - Added the platform metrics log. This log contains a listing of core Sensu metrics in influx-line format. It is enabled by default but can be disabled with the --disable-platform-metrics flag. By default the log is appended to every 60s, and written to /var/lib/sensu/sensu-backend/stats.log.

Reviewer: ```suggestion
- Added [platform metrics logging][] to log core Sensu metrics in InfluxDB Line Protocol format, along with the `disable-platform-metrics`, `platform-metrics-log-file`, and `platform-metrics-logging-interval` backend configuration flags for managing the platform metrics logging feature.
```
The link for this one is:
https://docs.sensu.io/sensu-go/6.5/observability-pipeline/observe-schedule/backend/#platform-metrics-logging

> - Open-sourced the previously enterprise-only event logger. The event logger can be used to send the events a backend processes to a rotatable log file.

Reviewer: ```suggestion
- [Event logging][] is no longer a commercial-only feature.
```

Link is https://docs.sensu.io/sensu-go/6.5/observability-pipeline/observe-schedule/backend/#event-logging

Reviewer: I think this event logging item should be moved up to the new features category

> - Added support for SENSU_BACKEND_ETCD_CLIENT_USERNAME and SENSU_BACKEND_ETCD_CLIENT_PASSWORD environment variables for connecting to external etcds that use user/password authentication instead of client certificate authentication. Typical with DBaaS etcd providers. These can only be set via these environment variables and intentionally cannot be set via flags.

Reviewer: ```suggestion
- Added environment variables `SENSU_BACKEND_ETCD_CLIENT_USERNAME` and `SENSU_BACKEND_ETCD_CLIENT_PASSWORD` for connecting to external etcd via username and password authentication instead of certificate authentication. There are no corresponding configuration flags &mdash; these configuration options must be set via environment variables.
```

Reviewer: I will have a link to add for this item, but I haven't created the PR yet

- Upgraded Go version from 1.16.5 to 1.17.1.

> - Added support for environment variable arguments in sensuctl.

Reviewer: I think this one snuck in from 6.4.1
```suggestion
```

> - ([Commercial feature][215]) Added support for pipelines from sensu-go.

Reviewer: commercial features should be listed first in each category.

- ([Commercial feature][215]) Added the pipeline/v1.TCPStreamHandler type.
- ([Commercial feature][215]) Added the SumoLogic metrics handler.

> - ([Commercial feature][215]) Added Prometheus metrics for TCPStreamHandler.

Reviewer: These should be moved up to the new features category

> - ([Commercial feature][215]) Added the pipeline/v1.TCPStreamHandler type.
> - ([Commercial feature][215]) Added the SumoLogic metrics handler.
> - ([Commercial feature][215]) Added Prometheus metrics for TCPStreamHandler.

Reviewer: ```suggestion
- ([Commercial feature][]) Added support for [Sumo Logic metrics handlers][] and [TCP stream handlers],. The [pipeline API][] provides HTTP access for retrieving and configuring Sumo Logic metrics handlers and TCP stream handlers.
- ([Commercial feature][]) Added Prometheus metrics for the TCP stream handlers:
    - sensu_go_tcp_stream_handler_events: Total number of events handled by the TCP stream handler
    - sensu_go_tcp_stream_handler_errors: The total number of errors produced by the TCP stream handler
    - sensu_go_tcp_stream_handler_latency: Distribution of handler latencies, in milliseconds, for the TCP stream handler
    - sensu_go_tcp_stream_handler_connection_acquisition_latency: Distribution of connection acquisition latencies (how long it takes to acquire a connection from the connection pool), in milliseconds, within the TCP stream handler
```

The commercial feature reference needs to be updated to `[]: /sensu-go/6.5/commercial/` for 6.5

**FIXES:**

> - Sensuctl env now properly displays the SENSU_API_KEY and SENSU_TIMEOUT environment variables.
> - Sensuctl command exec now properly adds the SENSU_API_KEY and SENSU_TIMEOUT variables to the command's environment.

Reviewer: ```suggestion
- sensuctl env now properly lists `SENSU_API_KEY` and `SENSU_TIMEOUT` as options for [exporting environment variables][]. In addition, sensuctl command exec now properly adds the `SENSU_API_KEY` and `SENSU_TIMEOUT` variables to the command's environment.
```

The link for this one is
https://docs.sensu.io/sensu-go/6.5/sensuctl/environment-variables/#export-environment-variables-with-sensuctl-env

> - ([Commercial feature][215]) Added support for pipelines from sensu-go.

Reviewer: ```suggestion
```

Reviewer: No need to list stuff that's already covered in Sensu Go.

> - Fixed a crash when running the backend on darwin/arm64 when compressing a wrapped resource.

Reviewer: ```suggestion
- Fixed a bug that could cause a crash when running the backend on darwin/arm64 and compressing a wrapped resource.
```

> - Fixed a bug where large number of silences could cause etcd errors by not exceeding etcd'd default maximum number of transaction operations.

Reviewer: ```suggestion
- Fixed a bug that could result in an etcd error if the number of silences in a given transaction exceeded etcd's default maximum number of operations per transaction. 
```

> 

Reviewer: Looks like 6.5.0 also has a security update, so we need a category listed after Improvements and before Fixes...something like this:

```
**SECURITY:**

- Migrated [dgrijalva/jwt-go][] to [golang-jwt/jwt][] to address a vulnerability that would allow attackers to bypass intended access restrictions in situations. Read [CVE-2020-26160][] for more information.
```

Also need to add these reference links at the bottom of the page, and then add the corresponding numbers in the empty `[]` in the release note item:
https://github.com/dgrijalva/jwt-go
https://github.com/golang-jwt/jwt
https://nvd.nist.gov/vuln/detail/CVE-2020-26160

> 

Reviewer: Commercial features should be first in the list. These are also all new features rather than improvements.

- ([Commercial feature][229]) Added Prometheus metrics for TCP stream handlers:
    - sensu_go_tcp_stream_handler_events: Total number of events handled by the TCP stream handler
    - sensu_go_tcp_stream_handler_errors: The total number of errors produced by the TCP stream handler
    - sensu_go_tcp_stream_handler_latency: Distribution of handler latencies, in milliseconds, for the TCP stream handler
    - sensu_go_tcp_stream_handler_connection_acquisition_latency: Distribution of connection acquisition latencies (how long it takes to acquire a connection from the connection pool), in milliseconds, within the TCP stream handler
- New [pipelines][233] resource allows you to specify event filters, mutators, and handlers in a single workflow instead of listing filters and mutators in handler definitions. You can reference pipelines in your check definitions. The [`/pipelines` API endpoint][234] provides HTTP access for retrieving pipeline data and configuring pipelines, and you can use [sensuctl][235] to manage pipelines.
- Check definitions now include the [pipelines attribute][236] for specifying pipeline resources to use for the check's observability events.

> - [Event logging][237] is no longer a commercial-only feature.

Reviewer: ```suggestion
- [Event logging][237] is no longer a commercial-only feature.
- You can now set sensuctl environment variables for a [single sensuctl command][243] or with [sensuctl configure][244].
```

Author: done

**IMPROVEMENTS:**

- Added [platform metrics logging][238] to log core Sensu metrics in InfluxDB Line Protocol format, along with the `disable-platform-metrics`, `platform-metrics-log-file`, and `platform-metrics-logging-interval` backend configuration flags for managing the platform metrics logging feature.

> - Added environment variables `SENSU_BACKEND_ETCD_CLIENT_USERNAME` and `SENSU_BACKEND_ETCD_CLIENT_PASSWORD` for connecting to external etcd via username and password authentication instead of certificate authentication. There are no corresponding configuration flags &mdash; these configuration options must be set via environment variables.

Reviewer: There will be a link to add for this item when the PR is approved (https://github.com/sensu/sensu-docs/pull/3398)

Read the [upgrade guide][1] for information about upgrading to the latest version of Sensu Go.

---
## 6.5.0 release notes

> **October 12, 2021** &mdash; The latest release of Sensu Go, version 6.5.0, is now available for download.

Reviewer: ```suggestion
**October 13, 2021** &mdash; The latest release of Sensu Go, version 6.5.0, is now available for download.
```
```suggestion
**October 12, 2021** &mdash; The latest release of Sensu Go, version 6.5.0, is now available for download.
```
```suggestion
**October 12, 2021** &mdash; The latest release of Sensu Go, version 6.5.0, is now available for download.
```

> Insert Release Synopsis.

Reviewer 2: ```suggestion
```

Reviewer: I think Yussuf is still waiting on the release synopsis from Sean

Read the [upgrade guide][1] to upgrade Sensu to version 6.5.0.

**NEW FEATURES:**

> - ([Commercial feature][229]) Added support for [Sumo Logic metrics handlers][230] and [TCP stream handlers][231]. The [pipeline API][232] provides HTTP access for retrieving and configuring Sumo Logic metrics handlers and TCP stream handlers.

Reviewer: ```suggestion
- ([Commercial feature][229]) Added [Sensu Plus][247], a built-in integration you can use to transmit your Sensu observability data to Sumo Logic via the Sumo Logic HTTP Logs and Metrics Source.
- ([Commercial feature][229]) Added support for [Sumo Logic metrics handlers][230] and [TCP stream handlers][231]. The [pipeline API][232] provides HTTP access for retrieving and configuring Sumo Logic metrics handlers and TCP stream handlers.
```

> - ([Commercial feature][229]) Added Prometheus metrics for TCP stream handlers:

Reviewer: ```suggestion
- ([Commercial feature][229]) You can now [view resource data][249] for events, entities, and configuration resources like checks and handlers directly in the web UI.
- ([Commercial feature][229]) In the web UI, you can [execute individual checks on demand][250] either according to existing subscriptions or on specific agents by adding and removing subscriptions without making changes to the saved check subscriptions.
- ([Commercial feature][229]) Added Prometheus metrics for TCP stream handlers:
```

    - sensu_go_tcp_stream_handler_events: Total number of events handled by the TCP stream handler
    - sensu_go_tcp_stream_handler_errors: The total number of errors produced by the TCP stream handler
    - sensu_go_tcp_stream_handler_latency: Distribution of handler latencies, in milliseconds, for the TCP stream handler
    - sensu_go_tcp_stream_handler_connection_acquisition_latency: Distribution of connection acquisition latencies (how long it takes to acquire a connection from the connection pool), in milliseconds, within the TCP stream handler

> - New [pipelines][233] resource allows you to specify event filters, mutators, and handlers in a single workflow instead of listing filters and mutators in handler definitions. You can reference pipelines in your check definitions. The [`/pipelines` API endpoint][234] provides HTTP access for retrieving pipeline data and configuring pipelines, and you can use [sensuctl][235] to manage pipelines.

Reviewer: ```suggestion
- New [pipelines][233] resource allows you to specify event filters, mutators, and handlers in a single workflow instead of listing filters and mutators in handler definitions. You can reference pipelines in your check definitions. The [`/pipelines` API endpoint][234] provides HTTP access for retrieving pipeline data and configuring pipelines, and you can use [sensuctl][235] to manage pipelines. [Upgrade your Sensu agents][252] to Sensu Go 6.5.0 to use pipelines resources.
```

> - Check definitions now include the [pipelines attribute][236] for specifying pipeline resources to use for the check's observability events.

Reviewer: ```suggestion
- [JavaScript mutators][246] are now available. JavaScript mutators are evaluated by the Otto JavaScript VM as JavaScript programs, which enables greater throughput at scale than pipe mutators.
- Check definitions now include the [pipelines attribute][236] for specifying pipeline resources to use for the check's observability events.
```

Reviewer 2: 🙏

> - [Event logging][237] is no longer a commercial-only feature.

Reviewer: ```suggestion
- Added [platform metrics logging][238] to log core Sensu metrics in InfluxDB Line Protocol format, along with the `disable-platform-metrics`, `platform-metrics-log-file`, and `platform-metrics-logging-interval` backend configuration flags for managing the platform metrics logging feature.
- [Event logging][237] is no longer a commercial-only feature.
```

- You can now set sensuctl environment variables for a [single sensuctl command][243] or with [sensuctl configure][244].

**IMPROVEMENTS:**

> - Added [platform metrics logging][238] to log core Sensu metrics in InfluxDB Line Protocol format, along with the `disable-platform-metrics`, `platform-metrics-log-file`, and `platform-metrics-logging-interval` backend configuration flags for managing the platform metrics logging feature.

Reviewer: ```suggestion
```

> - Added environment variables `SENSU_BACKEND_ETCD_CLIENT_USERNAME` and `SENSU_BACKEND_ETCD_CLIENT_PASSWORD` for connecting to external etcd via username and password authentication instead of certificate authentication. There are no corresponding configuration flags &mdash; these configuration options must be set via environment variables.

Reviewer: ```suggestion
- Added environment variables `SENSU_BACKEND_ETCD_CLIENT_USERNAME` and `SENSU_BACKEND_ETCD_CLIENT_PASSWORD` for [connecting to external etcd via username and password authentication][245] instead of certificate authentication. There are no corresponding configuration flags &mdash; these configuration options must be set via environment variables.
```

> - Upgraded Go version from 1.16.5 to 1.17.1.

Reviewer: ```suggestion
- You can now add an [API key][248] when you initialize the backend to make automated cluster setup and deployment more straightforward.
- Events now include the name of the agent that processed the event in the [`processed_by` attribute][251] to help you determine which agent processed an event executed by a proxy check request or a POST request to the events API.
- Added the [`ignore-already-initialized` backend flag][253], which you can use to suppress the "already initialized" response and return an exit code 0 if a cluster has already been initialized.
- Upgraded Go version from 1.16.5 to 1.17.1.
```

[226]: /sensu-go/6.4/observability-pipeline/observe-schedule/backend/#event-log-parallel-encoders
[227]: /sensu-go/6.4/api/metrics/
[228]: /sensu-go/6.4/observability-pipeline/observe-schedule/backend/#event-logging
[229]: /sensu-go/6.5/commercial/
[230]: /sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers/
[231]: /sensu-go/6.5/observability-pipeline/observe-process/tcp-stream-handlers/
[232]: /sensu-go/6.5/api/pipeline/
[233]: /sensu-go/6.5/observability-pipeline/observe-process/pipelines/
[234]: /sensu-go/6.5/api/pipeline-resource/
[235]: /sensu-go/6.5/sensuctl/create-manage-resources/#manage-resources
[236]: /sensu-go/6.5/observability-pipeline/observe-schedule/checks/#pipelines-attribute
[237]: /sensu-go/6.5/observability-pipeline/observe-schedule/backend/#event-logging
[238]: /sensu-go/6.5/observability-pipeline/observe-schedule/backend/#platform-metrics-logging
[239]: https://github.com/dgrijalva/jwt-go
[240]: https://github.com/golang-jwt/jwt
[241]: https://nvd.nist.gov/vuln/detail/CVE-2020-26160

> [242]: /sensu-go/6.5/sensuctl/environment-variables/#export-environment-variables-with-sensuctl-env

Reviewer: ```suggestion
[242]: /sensu-go/6.5/sensuctl/environment-variables/#export-environment-variables-with-sensuctl-env
[243]: /sensu-go/6.5/sensuctl/environment-variables/#set-environment-variables-for-a-single-command
[244]: /sensu-go/6.5/sensuctl/environment-variables/#set-environment-variables-with-sensuctl-configure
```

[227]: /sensu-go/6.4/api/metrics/
[228]: /sensu-go/6.4/observability-pipeline/observe-schedule/backend/#event-logging
[229]: /sensu-go/6.5/commercial/
[230]: /sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers/
[231]: /sensu-go/6.5/observability-pipeline/observe-process/tcp-stream-handlers/
[232]: /sensu-go/6.5/api/pipeline/
[233]: /sensu-go/6.5/observability-pipeline/observe-process/pipelines/
[234]: /sensu-go/6.5/api/pipeline-resource/
[235]: /sensu-go/6.5/sensuctl/create-manage-resources/#manage-resources
[236]: /sensu-go/6.5/observability-pipeline/observe-schedule/checks/#pipelines-attribute
[237]: /sensu-go/6.5/observability-pipeline/observe-schedule/backend/#event-logging
[238]: /sensu-go/6.5/observability-pipeline/observe-schedule/backend/#platform-metrics-logging
[239]: https://github.com/dgrijalva/jwt-go
[240]: https://github.com/golang-jwt/jwt
[241]: https://nvd.nist.gov/vuln/detail/CVE-2020-26160
[242]: /sensu-go/6.5/sensuctl/environment-variables/#export-environment-variables-with-sensuctl-env
[243]: /sensu-go/6.5/sensuctl/environment-variables/#set-environment-variables-for-a-single-command

> [244]: /sensu-go/6.5/sensuctl/environment-variables/#set-environment-variables-with-sensuctl-configure

Reviewer: ```suggestion
[245]: /sensu-go/6.5/operations/deploy-sensu/cluster-sensu/#authenticate-with-username-and-password-for-external-etcd
```

> [245]: /sensu-go/6.5/operations/deploy-sensu/cluster-sensu/#authenticate-with-username-and-password-for-external-etcd

Reviewer: ```suggestion
[245]: /sensu-go/6.5/operations/deploy-sensu/cluster-sensu/#authenticate-with-username-and-password-for-external-etcd
[246]: /sensu-go/6.5/observability-pipeline/observe-transform/mutators/#javascript-mutators
[247]: /sensu-go/6.5/sensu-plus/
[248]: /sensu-go/6.5/observability-pipeline/observe-schedule/backend/#add-api-key-for-initialization
[249]: /sensu-go/6.5/web-ui/view-manage-resources/#view-resource-data-in-the-web-ui
[250]: /sensu-go/6.5/web-ui/view-manage-resources/#execute-checks-on-demand
[251]: /sensu-go/6.5/observability-pipeline/observe-events/events/#processedby-attribute
[252]: /sensu-go/6.5/operations/maintain-sensu/upgrade/#upgrade-to-sensu-go-650-from-any-previous-version
[253]: /sensu-go/6.5/observability-pipeline/observe-schedule/backend/#initialization-ignore-already-initialized-flag
```

[227]: /sensu-go/6.4/api/metrics/
[228]: /sensu-go/6.4/observability-pipeline/observe-schedule/backend/#event-logging
[229]: /sensu-go/6.5/commercial/
[230]: /sensu-go/6.5/observability-pipeline/observe-process/sumo-logic-metrics-handlers/
[231]: /sensu-go/6.5/observability-pipeline/observe-process/tcp-stream-handlers/
[232]: /sensu-go/6.5/api/pipeline/
[233]: /sensu-go/6.5/observability-pipeline/observe-process/pipelines/
[234]: /sensu-go/6.5/api/pipeline-resource/
[235]: /sensu-go/6.5/sensuctl/create-manage-resources/#manage-resources
[236]: /sensu-go/6.5/observability-pipeline/observe-schedule/checks/#pipelines-attribute
[237]: /sensu-go/6.5/observability-pipeline/observe-schedule/backend/#event-logging
[238]: /sensu-go/6.5/observability-pipeline/observe-schedule/backend/#platform-metrics-logging
[239]: https://github.com/dgrijalva/jwt-go
[240]: https://github.com/golang-jwt/jwt
[241]: https://nvd.nist.gov/vuln/detail/CVE-2020-26160
[242]: /sensu-go/6.5/sensuctl/environment-variables/#export-environment-variables-with-sensuctl-env
[243]: /sensu-go/6.5/sensuctl/environment-variables/#set-environment-variables-for-a-single-command

> [245]: /sensu-go/6.5/operations/deploy-sensu/cluster-sensu/#authenticate-with-username-and-password-for-external-etcd

Reviewer: ```suggestion
[244]: /sensu-go/6.5/sensuctl/environment-variables/#set-environment-variables-with-sensuctl-configure
[245]: /sensu-go/6.5/operations/deploy-sensu/cluster-sensu/#authenticate-with-username-and-password-for-external-etcd
```
